x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -105,10 +105,15 @@ export const TransactionActionMenu: FunctionComponent<Props> = ({
 
             if (app === 'uptime' || app === 'metrics' || app === 'logs') {
               event.preventDefault();
+              let search = parsed.search || '';
+              if (parsed.hash) {
+                // convert hash to search. passing a hash is broken, either in uptime or in navigateToApp
+                search += parsed.hash.replace('#/?', search ? '&' : '?');
```

Comment:
This look odd - shouldn't we fix the link instead of changing it here?

Reply:
it does look odd, but I couldn't figure it out and given the time constraints I figured it's best to address it on our side for now. I'll investigate in a bit to see if it's an Uptime or platform issue.

Reply:
👍

Reply:
rather than reformatting the link on click, I'm just using `search` instead of `hash` when creating the link now. I've asked the Uptime team to verify whether that's safe (it seems to work).

Reply:
I'm now re-reading your comment and looks like that's what you were suggesting anyway 😅 I misread it as us needing to fix the underlying issue where using hashes in navigateToApp seems broken.